### ISSUE TITLE
TestClientTransport inherits from wrong base

### ISSUE THREAD

**Issue by devkral** [dymmond/lilya#90]
TestClientTransport is supposed to handle ASGI but inherits from httpx.BaseTransport which is for WSGI and handles with deprecation warnings ASGI.

It should inherit of httpx.AsyncBaseTransport and implement it.

**Comment by tarsil** [comment]
Yes that is correct. For the async client we adopt something else. In fact you are correct. We should implement the AsyncTestClient directly. 

We should also deprecate python 3.8 now
